### DISCUSSION TITLE
Flare Elemental/Magical Effects On Equipped Items?

### DISCUSSION THREAD

**Discussion by WithinAmnesia** [flareteam/flare-game#860] [Q&A] [answered]
How can a person add elemental / magical effects (such as elemental damage) to an equipped item (especially with weapons) so that item assets like this make more sense: https://opengameart.org/content/flare-item-variation-60x60-only ? Diablo II, vanilla WoW and Baldur's Gate all have elemental / magical effects on both equipped weapons and armour. Is a similar gameplay feature / item system of elemental / magical effects on equipped items possible within the Flare engine?

Can a weapon (or even any equipped item) have a 1-100% chance of casting a spell to simulate added elemental damage / add abilities to weapon swings / deal X amount of X type of damage to attacker(s) when struck in combat? Such as a ~'trinket' that adds a 1% chance to cast X spell / deal X amount of elemental damage to target on successful strike. Or something that makes 'magic' happen like a 'proc' / added bonus for equipped items. What can Flare do here for 'flashy magic bonuses' on equipped items within reason?

Can armour have ~'when struck abilities' that trigger and result in the attacker to take X elemental damage or trigger some kind of spell at the foot of the wearer (like a fire blast centered on the wearer)?

Also can equipped items have one use abilities with a cool down like once every ~5 minutes use X ability?

How can equipped items incorporate elemental / magical bonuses (damage, on hit etc.) within Flare?

All thoughts are welcome for I am really curious as to how to reasonably add elemental / magical effects / uses for equipped items to better make them feel and function as proper magical equipment within the gameplay experience.

**Comment by dorkster** [comment, marked as answer]
Without any engine changes, I think the way to go about this is by employing a similar method to slingshot/bow ammo. The Swing power would become a meta power so that the equipped weapon could swap out the "real" power. For the elemental versions of the Swing power, you could add a `post_power` that creates the elemental damage effect.

But ideally, I think elemental damages should become normal stats like melee/ranged/mental damages are. That way, items/powers could simply provide bonuses to those stats. Of course, you would still need to use the meta power method if you wanted to add visual effects.

>Can a weapon (or even any equipped item) have a 1-100% chance of casting a spell[...]?

Not currently. I honestly find those sorts of "%X chance to Y" effects boring, but I guess they could add some variety.

>Also can equipped items have one use abilities with a cool down like once every ~5 minutes use X ability?

Yes, equipped items can have usable powers. See the "DEV Boots", which grant the ability to teleport, as an example. The teleport from that example doesn't have a cooldown, but you could add `cooldown=300s` to give it a 5 minute cooldown.

**Reply by WithinAmnesia** [reply]
Thank you I will have to do some more testing.

Can an item apply a passive but visible buff like an aura / spell effect like a shield constantly? 

Either way is good to hear that there are methods / ways of giving elemental / magical effects to weapons and armour:-).

**Reply by dorkster** [reply]
>Can an item apply a passive but visible buff like an aura / spell effect like a shield constantly?

Yes. Here's basically what you'd need to do:

- Create the passive power first. For an aoe aura or shield, you may need to add a `post_power` to the passive that creates the desired effect.
- Add the passive power to the powers menu. You'll most likely want to hide the power by setting `visible=false`.
- Add the passive power to the item with the `power` attribute

**Reply by WithinAmnesia** [reply]
Okay that is good to hear.

I will have to make a bunch of test builds and see this all looks like running in the Flare engine. If I have say 200 melee talent tree abilities if I make ~200 elemental type weapons will I have to make 40,000 powers to fulfill every elemental weapon power to talent tree melee power combination?

Is there no way to make the 'Swing' ability (and all derivatives) have a check / ability to apply an equipped power effect? Or is there a way to create a power that acts as a passive aura that adds +X damage of an arbitrary type of element / magic? If so that could prevent needing to create ~20,000 power combinations and apply a passive power 'aura'. Although I might be missing a few critical steps to make that work currently.

My loose thoughts: Man having a way to trigger a post power effect on a weapon swing would be so creatively powerful to build weapons with. The same applies for armour with a 'thorns' effect for a way to trigger a post power effect on a player or enemy via being hit an applying that post power to either the player or the attacker or a general aoe. I remember in making new items in Baldur's Gate's Infinity Engine there was a way to apply a spell / ability trigger of different types like constant, activate anywhere on screen, targeted creature required, apply to self, apply to target, apply to friendly, etc. That made it so that if a weapon / armour had an effect it could be made to apply to a select type of target and in the case of weapons the spell effects could be say [Weapon: Add Effect > ~'Fire Damage 1 to 1 (does 1 fire damage) > ~Apply to Target > Chance 100%] and that would make the weapon's ability only trigger on a target 100% of the time and it would apply 1 fire damage on a successful hit.

The weapons could off load special abilities / powers and each item could work in a 'vacuum' and could apply its unique powers via being able to target the player's target and or aoe and an other check of if it worked on all, friendly or enemy. I am not sure how the targeting system works with the Flare engine but it would be very creatively empowering if there is a way to have abilities make a check for 'target of target' or something like that so what ever the creature (friendly or enemy) targets and triggers a power / spell / ability applied to ~'their target' (and or they cast an aoe effect) that the post power abilities can apply to that 'target of target' / whoever they are apply a power to.

I think WoW works similar to all of what I have said for that is tab targeting and the combat log looks like Baldur's Gate and Flare of who dealt what to what and by how much that constantly updates in sequential order. I am not 100% sure how this can be done and I hope there is a way to get the Flare engine's itemization (in terms of equipment special types of abilities / ~'post powers') nearly on par with games like Diablo, Baldur's Gate, Everquest, WoW / every new / big scale combat game with lots of items and abilities. I am currently trying to figure out how to close the engine capability / content creation scope ~'gap' and bring all of that depth on to the Flare engine.

This is so that this whole grand endeavour of just one person building efficiently with a commercially viable game engine (RPG Maker MV is trash and does not count :-P) feels that it can support the same kind of content found with the R.P.G. greats of Diablo, Baldur's Gate, Everquest and WoW that really adds huge amounts of depth and overall improved gameplay quality experience. This is a combat item driven exploration R.P.G. (or what ever Diablo II was called XD) and I really hope to be able to go toe to toe with these industry juggernauts as one person on my own terms with my creative and financial freedom intact being personally sovereign to my own destiny and succeed in creating one of the best R.P.G.s of all time. Aim high, fight hard, be unyielding.

There must be a way to have hundreds of talent tree abilities and a way to off load special item abilities / powers and have each item able to work in a 'vacuum'; just like Diablo, Baldur's Gate, Everquest, WoW etc... I really hope the Flare engine can support these amazing weapon / combat / itemization features. I sincerely hope to work towards building and properly releasing and supporting and expanding an R.P.G. similar to the classic great R.P.G.s that can stand to last the test time. Yet I have to find a way to have the ability to create versions of all the aspects that attributed to improving the player gameplay experience. There just has to be a way forward to support this amazing staple of magical weapons and armour and equipment in this amazing game engine. How amazing the combat and player goals and gameplay experience could be if this classic medieval fantasy feature can be successfully ported / implemented into the Flare engine.

It seems so hard to find great creative tools over the past decades to work with. For once they get made they are seemingly never released. It is kind of a god send to get the open source Flare engine and every one here who adds to it year after year. I am really hoping I can take a serious risk this half decade and with the Flare engine put the work in and go the next step further and compete toe to toe with the great classic R.P.G.s and win. To make a high quality gameplay driven experience in the form of a video game that can be commercially successful ~'enough' to justify the thousands of hours of work. Moreover to live and grow and struggle and battle up from nothing and by the grace of god and the community and tens of thousands of hours of work over decades of failing upwards towards the same lofty lifelong ambition. To fulfill a dream of mine that haunts myself to my very soul every single day after day to become an honest to god proper video game developer and make great games where the player always wins first. Bit by bit, step by step; iterate, iterate, iterate.
